src/app/components/QRCode/index.tsx

```diff
@@ -15,21 +18,56 @@ export type Props = {
   // (meaning you have to aim your phone very precisely and have to wait longer for the reader
   // to recognize the QR code)
   level?: "Q" | undefined;
+  onCopy?: () => void;
 };
 
-export default function QRCode({ value, size, level, className }: Props) {
+export default function QRCode({
+  value,
+  size,
+  level,
+  className,
+  onCopy,
+}: Props) {
   const theme = useTheme();
   const fgColor = theme === "dark" ? "#FFFFFF" : "#000000";
   const bgColor = theme === "dark" ? "#000000" : "#FFFFFF";
+  const { t } = useTranslation("components");
+  const [isHovering, setIsHovering] = useState(false);
+
+  const handleCopy = () => {
+    navigator.clipboard.writeText(value);
+    onCopy?.();
+  };
 
   return (
-    <ReactQRCode
-      value={value}
-      size={size}
-      fgColor={fgColor}
-      bgColor={bgColor}
-      className={classNames("w-full h-auto rounded-md", className ?? "")}
-      level={level}
-    />
+    <div
+      className="relative cursor-pointer"
+      onClick={handleCopy}
+      onMouseEnter={() => setIsHovering(true)}
+      onMouseLeave={() => setIsHovering(false)}
+    >
+      <ReactQRCode
+        value={value}
+        size={size}
+        fgColor={fgColor}
+        bgColor={bgColor}
+        className={classNames(
+          "w-full h-auto rounded-md transition-opacity",
+          isHovering ? "opacity-80" : "opacity-100",
```

Comment:
for hover why we need to use state varbiles? can use tailwind here only? 

eg. 
```className="text-gray-600 hover: text-gray-800"```